### DISCUSSION TITLE
how to make open-mcr work for other omr's

### DISCUSSION THREAD

**Discussion by sricharan22** [iansan5653/open-mcr#97] [General]
Hi,
I have created a new OMR of my own and would like to change the code to make it work for that specific structure. Is there any documentation of steps that I need to follow?

**Comment by iansan5653** [comment]
Hi @sricharan22 . I've migrated this discussion to the open-mcr repository. I am not actively maintaining the open-mcr project, and creating new form sheets is not something that is supported. However, if you are willing to fork the code to make this work, you could do roughly the following:

1. Design a new sheet:
    1. Copy an existing sheet and maintain the same corner marks
    2. Keep all the circles aligned to the same grid
    3. Build 'regions' of marks with sets of groups. The supported regions are declared here:  https://github.com/iansan5653/open-mcr/blob/2d54ed846699b6f51ce5efa0284cb1961c2369e7/src/grid_info.py#L12
2. Construct a new instance of `FormVariant` here: https://github.com/iansan5653/open-mcr/blob/master/src/grid_info.py. This is where you would provide the information about your new sheet such as where the regions are located and how large they are
3. If using the command line interface, update the `form_variant` here to use your form: https://github.com/iansan5653/open-mcr/blob/2d54ed846699b6f51ce5efa0284cb1961c2369e7/src/main.py#L67
4. If using the GUI, update `form_variant` here: https://github.com/iansan5653/open-mcr/blob/2d54ed846699b6f51ce5efa0284cb1961c2369e7/src/main_gui.py#L23
